Dockerfile

```diff
@@ -3,6 +3,10 @@
 # https://github.com/docker-library/repo-info/tree/master/repos/alpine/remote
 FROM alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad
 ARG EXTENSION_FILE
+ARG COMMIT_TIMESTAMP
 COPY ${EXTENSION_FILE} /opt/elastic-apm-extension
-COPY /NOTICE.txt /opt/NOTICE.txt
-COPY /dependencies.asciidoc /opt/dependencies.asciidoc
+COPY NOTICE.txt dependencies.asciidoc /opt/
+
+# Related to reproducible builds
+RUN find /opt -exec touch -amdt "${COMMIT_TIMESTAMP}" {} \;
```

Comment:
This is silently failing.

Adding `--debug` to the gorelease command in the make task shows:

```
    • #8 0.264 touch: invalid date 't'               cmd=[docker buildx] cwd=dist/goreleaserdocker3058620621
    • #8 0.265 touch: invalid date 't'               cmd=[docker buildx] cwd=dist/goreleaserdocker3058620621
    • #8 0.265 touch: invalid date 't'               cmd=[docker buildx] cwd=dist/goreleaserdocker3058620621
    • #8 0.266 touch: invalid date 't'               cmd=[docker buildx] cwd=dist/goreleaserdocker3058620621
```

Opening a shell into an alpine container: 

```
$ docker run -it alpine:3.16.3 /bin/sh
/ # touch -h
BusyBox v1.35.0 multi-call binary.

Usage: touch [-cham] [-d DATE] [-t DATE] [-r FILE] FILE...

Update mtime of FILEs

        -c      Don't create files
        -h      Don't follow links
        -a      Change only atime
        -m      Change only mtime
        -d DT   Date/time to use
        -t DT   Date/time to use
        -r FILE Use FILE's date/time
```

The correct command should be `RUN find /opt -exec touch -am -d "${COMMIT_TIMESTAMP}" -t "${COMMIT_TIMESTAMP}" {} \;`

but then `touch` complains about the date format being invalid.

Previously we used `date` to create a posix compliant task (https://github.com/elastic/apm-aws-lambda/pull/362/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L11-L15) but since the command is run inside an alpine container we can assume busybox and tailor the syntax.

The final command should be something like:

```suggestion
RUN find /opt -exec touch -am -d $(date -u -d @"${COMMIT_TIMESTAMP}" "+%Y%m%d%H%M.%S") -t $(date -u -d @"${COMMIT_TIMESTAMP}" "+%Y%m%d%H%M.%S") {} \;
```

Reply:
Thank you very much for this.

Reply:
https://github.com/elastic/apm-aws-lambda/pull/362/commits/daa0ceec202377c22d98238e4b84d67088b135e1